src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -902,59 +899,97 @@ private void ProcessDependencies(PackageBuilder builder)
             }
         }
 
-        private void AddDependencies(Dictionary<String, Tuple<IPackage, NuGet.PackageDependency>> packagesAndDependencies)
+        private bool FindDependency(PackageIdentity projectPackage, IEnumerable<Tuple<PackageArchiveReader, PackageDependency>> packagesAndDependencies)
+        {
+            // returns true if the dependency should be added to the package
+            // This happens if the dependency is not a dependency of a dependecy
+            // Or if the project dependency version is > the dependency's dependency version
+            bool found = false;
+            foreach (var reader in packagesAndDependencies)
+            {
+                foreach (var set in reader.Item1.GetPackageDependencies())
+                {
+                    foreach (var dependency in set.Packages)
+                    {
+                        if (dependency.Id.Equals(projectPackage.Id, StringComparison.OrdinalIgnoreCase))
+                        {
+                            found = true;
+
+                            if (dependency.VersionRange.MinVersion < projectPackage.Version ||
+                                (!dependency.VersionRange.IsMinInclusive &&
+                                dependency.VersionRange.MinVersion == projectPackage.Version))
+                            {
+                                return true;
```

Comment:
I thought the fix was that only if the project required a version later than it's dependency requires should it include it.  If the project has an earlier minimum, it doesn't matter as much. At the same time, it might not hurt to include anytime it's different.  What do you think?